File: LibraryApp/app/src/main/java/com/example/libraryapp/LostFoundAdapter.java
package com.example.libraryapp;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class LostFoundAdapter extends RecyclerView.Adapter<LostFoundAdapter.ViewHolder> {

    private List<LostItem> itemList;
    private OnItemClickListener listener;

    public interface OnItemClickListener {
        void onClaimClick(LostItem item);
    }

    public LostFoundAdapter(List<LostItem> itemList, OnItemClickListener listener) {
        this.itemList = itemList;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_lost_found, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        LostItem item = itemList.get(position);
        holder.desc.setText(item.getDescription());
        holder.date.setText("Found: " + item.getDateFound());
        holder.location.setText("Loc: " + item.getLocationFound());

        if ("Claim Pending".equals(item.getStatus())) {
            holder.btnClaim.setText("PENDING");
            holder.btnClaim.setEnabled(false);
        } else {
            holder.btnClaim.setText("CLAIM");
            holder.btnClaim.setEnabled(true);
            holder.btnClaim.setOnClickListener(v -> listener.onClaimClick(item));
        }
    }

    @Override
    public int getItemCount() { return itemList.size(); }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView desc, date, location;
        Button btnClaim;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            desc = itemView.findViewById(R.id.tvDescription);
            date = itemView.findViewById(R.id.tvDate);
            location = itemView.findViewById(R.id.tvLocation);
            btnClaim = itemView.findViewById(R.id.btnClaim);
        }
    }
}

File: src/main/java/com/example/libraryapp20/LostFoundAdapter.kt
package com.example.libraryapp20

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class LostFoundAdapter(
    private val itemList: List<LostItem>,
    private val listener: OnItemClickListener
) : RecyclerView.Adapter<LostFoundAdapter.ViewHolder>() {

    interface OnItemClickListener {
        fun onClaimClick(item: LostItem)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_lost_found, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = itemList[position]
        holder.desc.text = item.description ?: ""
        holder.date.text = "Found: ${item.dateFound ?: ""}"
        holder.location.text = "Loc: ${item.locationFound ?: ""}"

        if (item.status == "Claim Pending") {
            holder.btnClaim.text = "PENDING"
            holder.btnClaim.isEnabled = false
            holder.btnClaim.setOnClickListener(null)
        } else {
            holder.btnClaim.text = "CLAIM"
            holder.btnClaim.isEnabled = true
            holder.btnClaim.setOnClickListener {
                val pos = holder.bindingAdapterPosition
                if (pos != RecyclerView.NO_POSITION) {
                    listener.onClaimClick(itemList[pos])
                }
            }
        }
    }

    override fun getItemCount(): Int = itemList.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val desc: TextView = itemView.findViewById(R.id.tvDescription)
        val date: TextView = itemView.findViewById(R.id.tvDate)
        val location: TextView = itemView.findViewById(R.id.tvLocation)
        val btnClaim: Button = itemView.findViewById(R.id.btnClaim)
    }
}
